sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/TransientIOErrorsRetryingIterator.scala

```diff
@@ -46,9 +50,9 @@ private class TransientIOErrorsRetryingIterator
   // scalastyle:on null
   private val retryCount = new AtomicLong(0)
 
-  private[spark] var currentFeedResponseIterator: Option[Iterator[FeedResponse[ObjectNode]]] = None
-  private[spark] var currentItemIterator: Option[Iterator[ObjectNode]] = None
-
+  private[spark] var currentFeedResponseIterator: Option[BufferedIterator[FeedResponse[ObjectNode]]] = None
+  private[spark] var currentItemIterator: Option[BufferedIterator[ObjectNode]] = None
+  private val lastPagedFlux = new AtomicReference[Option[CosmosPagedFlux[ObjectNode]]](None)
```

Comment:
will the close() method called multiple times?  (that is why we need to track the lastPagedFlux?)

Reply:
no - i want to cancel even when moving from one flux to another in retries